### ISSUE TITLE
Possible bugs in postgres ast 

### ISSUE THREAD

**Issue by hanyisong** [sqlancer/sqlancer#630]
https://github.com/sqlancer/sqlancer/blob/master/src/sqlancer/postgres/ast/PostgresBinaryArithmeticOperation.java#L65

Why convert an expression to an integer?
case:
```
The operator is '+'
leftVal = 0.1
rightVal = 0.01
Then convert its result to boolean type, expected to return true, actually returned false
```

**Comment by hanyisong** [comment]
cc @mrigger

**Comment by mrigger** [comment]
I'm definitely confused by the method name and the class name. This class seems to apply a binary arithmetic operator on ints.

Are you sure that incorrect behavior is triggered? I just see the class being used at https://github.com/sqlancer/sqlancer/blob/c61d9a95dd3ec5dd6105bc8f240db9d42cbdb8cd/src/sqlancer/postgres/gen/PostgresExpressionGenerator.java#L443, which generates integer expressions.

**Comment by hanyisong** [comment]
> I'm definitely confused by the method name and the class name. This class seems to apply a binary arithmetic operator on ints.
> 
> Are you sure that incorrect behavior is triggered? I just see the class being used at
> 
> https://github.com/sqlancer/sqlancer/blob/c61d9a95dd3ec5dd6105bc8f240db9d42cbdb8cd/src/sqlancer/postgres/gen/PostgresExpressionGenerator.java#L443
> 
> , which generates integer expressions.

This is just a guess on my part, I didn't realize I was wrong until I saw that it only applies to integers.

Why not make it support float?

**Comment by mrigger** [comment]
Yes, we could support more data types in principle. Getting exactly the same results is a bit tricky for floating-point numbers, though.
